tutorials/lesson3-anatomy/coin.template.qnt

```diff
@@ -0,0 +1,953 @@
+            
+            <block>
+
+If you lucky, it fails right away. If it does not fail, run it multiple times,
+until it fails. To see, why it failed, evaluate `state` after executing
+the test. Do you understand why our hypothesis was wrong?
+
+**Exercise:** Fix the condition in `mintTwiceThenSendTest`,
+so the test never fails.
+
+If you carefully look at `mintTwiceThenSendTest`, you will see that it is still
+a single data point test, though the data point (the inputs) are chosen
+non-deterministically every time we run the test. In fact, REPL implements
+non-determinism via random choice.
+
+If you do not want to sit the whole day and run the test, you could integrate
+it into continuous integration, so it runs from time to time for different
+inputs. Also, we had to fix the sequence of actions in our test. We can do
+better with Quint.
+
+            </block>
+        </description>
+        <file>$file</file>
+        <line>$line</line>
+      </step>
+    !*/
+
+    // to run the random simulator for 10000 executions, each up to 10 actions,
+    // execute the following in REPL:
+    // _test(10000, 10, "init", "step", "totalSupplyDoesNotOverflowInv")
+    /*!
+      <step>
+        <title>Testing with non-deterministic inputs and control</title>
+        <description>
+
+It is often hard to find a good sequence of actions that breaks an invariant.
+Similar to how we let the computer to find the right inputs, we can use the
+computing power to look for bad sequences of actions.
+
+Quint REPL actually supports random search for sequences of actions that violate
+an invariant. Its UX is a bit ad hoc at the moment. We will improve it in the
+future. You can try it right away:
+
+            <run target="markdown">echo '_test(10000, 10, "init", "step", "totalSupplyDoesNotOverflowInv")' | quint -r coin.qnt::Coin</run>
+            <run target="codetour">echo '_test(10000, 10, "init", "step", "totalSupplyDoesNotOverflowInv")' | quint -r ./lesson3-anatomy/coin.qnt::Coin</run>
+
+            <block>
+
+The above command in REPL randomly produces sequences, starting with `init`
+and continuing with `step`, up to 10 actions. It checks the invariant
+`totalSupplyDoesNotOverflowInv` after every step. If the invariant is violated,
+the random search stops and returns `false`. If no invariant violaion is found,
+the search returns `true` after enumerating the specified number of runs,
+which is 10000 in our example.
+
+In our example, the search often finds an error. If it did not find an error,
+run the search again. Once `_test` returned `false`, you can print the trace
+leading to the bad state by evaluating `_lastTrace` in REPL.
+
+            </block>
+
+        </description>
+        <file>$file</file>
+        <line>$line</line>
+      </step>
+    !*/
+
+    /*!
+      <step>
+        <title>Does random testing always find bugs?</title>
+        <description>
```

Comment:
Code snippet is empty

Reply:
Oh, it's empty in markdown. Good point.

Reply:
I've fixed the script to omit empty code snippets